semgrep-core/src/cli/Main.ml

```diff
@@ -664,22 +664,18 @@ let iter_files_and_get_matches_and_exn_to_errors f files =
                 * Timeout here to give a better diagnostic.
                 *)
                | (Main_timeout _ | Out_of_memory) as exn ->
-                   let error_fun =
-                     match !Match_patterns.last_matched_rule with
-                     | None -> E.mk_error_no_rule
-                     | Some rule ->
-                         logger#info "critical exn while matching ruleid %s"
-                           rule.MR.id;
-                         logger#info "full pattern is: %s"
-                           rule.MR.pattern_string;
-                         E.mk_error rule.MR.id
```

Comment:
why we don't return an mk_error ~rule_id in that case then? Seems like we have a current rule, so we could be more precise?

Reply:
This is the mini-rule id, so it's not very useful to a user. Also, if semgrep-core ever times out while running a rule, it should be caught in `Run_rules.check`. If the exception reaches Main, there shouldn't be a rule_id